### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by tungbinhdinh89**
[https://youtu.be/CQbf70zbnWg](url)

### CODE REVIEW

File: src/megax/MegaApp/Vite/src/lib/apis/axios.instance.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    headers: {
+        "Content-type": "application/json",
+        Accept: "application/json",
+    },
+});
+
+api.interceptors.request.use(function (config) {
+    const token = storage.get("token");
+    console.log("token", token);
```
Comment (oclockvn) [inline src/megax/MegaApp/Vite/src/lib/apis/axios.instance.ts:14]: remove console log

File: src/megax/MegaApp/Vite/src/lib/models/login.model.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface UserLoginResponse {
+    token: string,
+    refreshToken: string,
+    expiryTime: Date,
+}
+
+export interface Result<T> {
```
Comment (oclockvn) [inline src/megax/MegaApp/Vite/src/lib/models/login.model.ts:7]: let's put this `Result` to `common.model.ts`

File: src/megax/MegaApp/Vite/src/pages/Login/LoginPage.tsx [hunk 3]
```diff
@@ -1,63 +1,53 @@
...
+  }, [isAuthenticated]);
 
   const {
     register,
     handleSubmit,
     formState: { errors },
-  } = useForm();
+  } = useForm<LoginFormType>();
+
+  const appDispatch = useAppDispatch();
```
Comment (oclockvn) [inline src/megax/MegaApp/Vite/src/pages/Login/LoginPage.tsx:36]: declare hooks in the top of function

File: src/megax/MegaApp/Vite/src/pages/Logout/Logout.tsx
Comment (oclockvn) [inline src/megax/MegaApp/Vite/src/pages/Logout/Logout.tsx:1]: no need this component

File: src/megax/MegaApp/Vite/src/provider/authProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+const AuthProvider = ({ children }) => {
+  const [token, setToken_] = useState(localStorage.getItem("token"));
+
+  const setToken = newToken => {
+    setToken_(newToken);
+  };
+
+  useEffect(() => {
+    if (token) {
+      axios.defaults.headers.common["Authorization"] = "Bearer " + token;
```
Comment (oclockvn) [inline src/megax/MegaApp/Vite/src/provider/authProvider.tsx:15]: no need this, we already have interceptor

**Review by oclockvn — CHANGES_REQUESTED** [review]

**Review by oclockvn — APPROVED** [review]

File: src/megax/MegaApp/Vite/src/provider/authProvider.tsx (hunk 4 above)
Comment (tungbinhdinh89) [reply src/megax/MegaApp/Vite/src/provider/authProvider.tsx]: oke

File: src/megax/MegaApp/Vite/src/pages/Logout/Logout.tsx
Comment (tungbinhdinh89) [reply src/megax/MegaApp/Vite/src/pages/Logout/Logout.tsx:1]: yes, i deleted

**Pull request merged**